platforms/nuttx/src/bootloader/common/crypto.c

```diff
@@ -66,4 +66,20 @@ bool decrypt_app(uint16_t idx, const image_toc_entry_t *toc_entries)
 	return false;
 }
 
+bool verify_boot_image(uint8_t key_index, void *boot_image, size_t image_len)
```

Comment:
Please try not to modify common/crypto.c;

An alternative for this is to just create a pair of image_toc_entry's (one for image and another for signature), and call verify_app(0, my_toc_entry). This can then be wrapped into a function "verify_boot_image" function in microchip/mpfs/main.c

Reply:
verify_boot_image removed from crypto.c and functionality moved to microchip/mpfs/main.c